src/node_url_pattern.cc

```diff
@@ -466,10 +466,20 @@ URLPattern::URLPatternOptions::FromJsObject(Environment* env,
   return options;
 }
 
+// Perform value lookup and cache the result in a v8::Global.
 #define URL_PATTERN_COMPONENT_GETTERS(uppercase_name, lowercase_name)          \
-  MaybeLocal<Value> URLPattern::uppercase_name() const {                       \
-    auto context = env()->context();                                           \
-    return ToV8Value(context, url_pattern_.get_##lowercase_name());            \
+  MaybeLocal<Value> URLPattern::uppercase_name() {                             \
+    auto isolate = env()->isolate();                                           \
+    if (lowercase_name.IsEmpty()) {                                            \
+      Local<Value> value;                                                      \
+      if (ToV8Value(env()->context(), url_pattern_.get_##lowercase_name())     \
+              .ToLocal(&value)) {                                              \
+        lowercase_name.Reset(isolate, value);                                  \
+        return value;                                                          \
+      }                                                                        \
+      return {};                                                               \
+    }                                                                          \
+    return lowercase_name.Get(isolate);                                        \
```

Comment:
If we're adding these then the MemoryInfo method should also be updated so that heap snapshots appropriately account for the held globals. You can find some examples by searching for `MemoryTracker`.

Reply:
Thanks @jasnell! I am not familiar with much of the memory tracking concepts, so would you mind reviewing if I added that correctly?

Thank you!

Reply:
I believe that is correct, yes. I'll do a more thorough review after the other PR lands and this is marked "Ready for review"

Reply:
Appreciate it, thanks for taking a look!